src/common/snippets/src/lowered/pass/solve_buffer_memory.cpp

```diff
@@ -14,73 +15,144 @@ namespace snippets {
+                const auto consumer_order = static_cast<int>(ov::snippets::pass::GetTopologicalOrder(consumer.get_expr()->get_node()));
+                e_finish = std::max(e_finish, consumer_order);  // the last consumer
             }
-            e_start = e_finish;
-
-            const auto& buffer_ins = buffer_expr->get_input_port_connectors();
-            for (const auto& buffer_in : buffer_ins) {
-                const auto& source = buffer_in->get_source();
-                e_start = static_cast<int>(ov::snippets::pass::GetTopologicalOrder(source.get_expr()->get_node()));
-
-                const auto buffer_siblings = buffer_in->get_consumers();
-                for (const auto& sibling : buffer_siblings) {
-                    if (const auto loop_end = ov::as_type_ptr<ov::snippets::op::LoopEnd>(sibling.get_expr()->get_node())) {
-                        e_start = std::min(e_start, static_cast<int>(ov::snippets::pass::GetTopologicalOrder(loop_end->get_loop_begin())));
-                    }
+        }
+        e_start = e_finish;
+
+        const auto& buffer_ins = buffer_expr->get_input_port_connectors();
+        for (const auto& buffer_in : buffer_ins) {
+            const auto& source = buffer_in->get_source();
+            e_start = static_cast<int>(ov::snippets::pass::GetTopologicalOrder(source.get_expr()->get_node()));
+
+            const auto buffer_siblings = buffer_in->get_consumers();
+            for (const auto& sibling : buffer_siblings) {
+                if (const auto loop_end = ov::as_type_ptr<ov::snippets::op::LoopEnd>(sibling.get_expr()->get_node())) {
+                    e_start = std::min(e_start, static_cast<int>(ov::snippets::pass::GetTopologicalOrder(loop_end->get_loop_begin())));
                 }
             }
-            OPENVINO_ASSERT(e_start <= e_finish, "Incorrect life time of buffer!");
+        }
+        OPENVINO_ASSERT(e_start <= e_finish, "Incorrect life time of buffer!");
 
-            auto buffer_size = static_cast<int64_t>(buffer->get_byte_size());
-            box_size = std::max(buffer_size, box_size);
+        auto buffer_size = static_cast<int64_t>(buffer->get_byte_size());
+        box.size = std::max(buffer_size, box.size);
 
-            box.start = std::min(e_start, box.start);
-            box.finish = std::max(e_finish, box.finish);
-        }
+        box.start = std::min(e_start, box.start);
+        box.finish = std::max(e_finish, box.finish);
+    }
+
+    std::vector<ov::MemorySolver::Box> boxes(map_boxes.size());
+    for (const auto& p : map_boxes) {
+        const auto& buffer_id = static_cast<size_t>(p.first);
+        OPENVINO_ASSERT(buffer_id < boxes.size(), "Incorrect Buffer Cluster ID");
+        boxes[buffer_id] = p.second;
 
         // We use data alignment to put data in the line cache
-        box.size = utils::div_up(box_size, m_alignment);
-        boxes.push_back(box);
+        boxes.at(buffer_id).size = utils::div_up(boxes.at(buffer_id).size, m_alignment);
```

Comment:
Have you checked that it has any effect on performance? 🤔 
Because the whole line will be loaded anyway, no matter if you `div_up` or not.

Reply:
I agree))))
Need to come back to this model in memory manager activity

Reply:
Added todo in commit https://github.com/openvinotoolkit/openvino/pull/24646/commits/a9fb56e5b7e051ecaa58b2f7f73d84ba0d698d28

Reply:
Nope, this must be a mistake 🙂 
this change is not in the commit (which is fine)